pymeasure/instruments/lecroy/lecroyT3DSO1204.py

```diff
@@ -0,0 +1,1324 @@
+            - "trigger_level": trigger level (float)
+            - "trigger_level2": trigger lower level for SLEW or RUNT trigger (float)
+            - "trigger_slope": trigger slope can be "negative" "positive" "window" (str)
+        """
+
+        ch_setup = {
+            "channel": self.number,
+            "attenuation": self.probe_attenuation,
+            "bandwidth_limit": self.bwlimit,
+            "coupling": self.coupling,
+            "offset": self.offset,
+            "skew_factor": self.skew_factor,
+            "display": self.display,
+            "unit": self.unit,
+            "volts_div": self.scale,
+            "inverted": self.invert,
+            "trigger_coupling": self.trigger_coupling,
+            "trigger_level": self.trigger_level,
+            "trigger_level2": self.trigger_level2,
+            "trigger_slope": self.trigger_slope
+        }
+        return ch_setup
+
+
+class LeCroyT3DSO1204(Instrument):
+    """ Represents the LeCroy T3DSO1204 Oscilloscope interface for interacting with the instrument.
+
+    Refer to the LeCroy T3DSO1204 Oscilloscope Programmer's Guide for further details about
+    using the lower-level methods to interact directly with the scope.
+
+    Attributes:
+        WRITE_INTERVAL_S: minimum time between two commands. If a command is received less than
+        WRITE_INTERVAL_S after the previous one, the code blocks until at least WRITE_INTERVAL_S
+        seconds have passed.
+        Because the oscilloscope takes a non neglibile time to perform some operations, it might
+        be needed for the user to tweak the sleep time between commands.
+        The WRITE_INTERVAL_S is set to 10ms as default however its optimal value heavily depends
+        on the actual commands and on the connection type, so it is impossible to give a unique
+        value to fit all cases. An interval between 10ms and 500ms second proved to be good,
+        depending on the commands and connection latency.
+
+    .. code-block:: python
+
+        scope = LeCroyT3DSO1204(resource)
+        scope.autoscale()
+        ch1_data_array, ch1_preamble = scope.download_waveform(source="C1", points=2000)
+        # ...
+        scope.shutdown()
+    """
+
+    _BOOLS = {True: "ON", False: "OFF"}
+
+    WRITE_INTERVAL_S = 0.01  # seconds
+
+    def __init__(self, adapter, **kwargs):
+        super().__init__(adapter, "LeCroy T3DSO1204 Oscilloscope", **kwargs)
+        if self.adapter.connection is not None:
+            self.adapter.connection.timeout = 3000
+        self._grid_number = 14  # Number of grids in the horizontal direction
+        self._seconds_since_last_write = 0  # Timestamp of the last command
+        self._header_size = 16  # bytes
+        self._footer_size = 2  # bytes
+        self.ch1 = Channel(self, 1)
+        self.ch2 = Channel(self, 2)
+        self.ch3 = Channel(self, 3)
+        self.ch4 = Channel(self, 4)
+        self.waveform_source = "C1"
+        self.default_setup()
```

Comment:
With the new Channels, you can define it that way:
```suggestion
    channels = Instrument.ChannelCreator(Channel, (1, 2, 3, 4))

    def __init__(self, adapter, **kwargs):
        super().__init__(adapter, "LeCroy T3DSO1204 Oscilloscope", **kwargs)
        if self.adapter.connection is not None:
            self.adapter.connection.timeout = 3000
        self._grid_number = 14  # Number of grids in the horizontal direction
        self._seconds_since_last_write = 0  # Timestamp of the last command
        self._header_size = 16  # bytes
        self._footer_size = 2  # bytes
        self.waveform_source = "C1"
        self.default_setup()
```

That way the channels will be created at initialization. However, the names will be "ch_1", "ch_2"... Which you have to change throughout your work.

To keep your names, you could add the parameter `prefix="ch"`, but please do so only, if really necessary.

Reply:
Same as the previous comment.

Reply:
Ok, as mentioned before, it would be good to name channels `ch_1` instead of `ch1` as you have them right now. Would you mind renaming them yet?

Reply:
I think that I implemented all necessary changes now

Reply:
I made a comment in the code. Resolving here.